modelmapper/cleaner.py

```diff
@@ -0,0 +1,204 @@
+            return result
+
+        for i, item in enumerate(field_values):
+            original_item = item
+            item = item.strip().lower()
+            if is_string:
+                if len(item) > max_string_len:
+                    raise ValueError(f'There is a value that is longer than {max_string_len} for {field_name}: {item}')
+
+            if is_integer or is_decimal:
+                item = normalize_numberic_values(item)
+
+            if is_nullable:
+                item = _mark_nulls(item)
+
+            if item is not None:
+                if is_boolean:
+                    item = _mark_booleans(item)
+
+                if is_integer or is_decimal or is_dollar or is_percent:
+                    try:
+                        item = Decimal(item)
+                    except Exception as e:
+                        raise TypeError(f'Unable to convert {item} into decimal: {e}') from None
+
+                if is_dollar:
+                    item = item * ONE_HUNDRED
+                if is_percent:
+                    item = item / ONE_HUNDRED
+                if is_integer:
+                    item = int(item)
+                if is_datetime:
+                    item_chars = set(item)
+                    if not item_chars <= self.settings.datetime_allowed_characters:
+                        if original_content_type == 'xls' and item_chars <= FLOAT_ACCEPTABLE:
+                            _format = original_content_type
+                            continue
+                        else:
+                            raise ValueError(f"Datetime value of {item} in {field_name} has characters "
+                                             "that are NOT defined in datetime_allowed_characters")
+                    msg = (f"{field_name} has invalid datetime format for {item} "
+                           f"that is not in {field_info.get('datetime_formats')}")
+                    try:
+                        _format = datetime_formats[-1]
+                        strptime(item, _format)
+                    except IndexError:
+                        raise ValueError(msg) from None
+                    except ValueError:
+                        if datetime_formats:
+                            datetime_formats.pop()
+                        else:
+                            raise ValueError(msg) from None
+                if is_string:
+                    item = original_item
+
+            field_values[i] = item
+
+        if is_datetime:
+            if _format == 'xls':
+                def xls_date(x):
+                    return xldate_as_datetime(float(x), self.xls_date_mode)
+                field_values[:] = map(lambda x: None if x is None else xls_date(x), field_values)
+            else:
+                field_values[:] = map(lambda x: None if x is None else strptime(x, _format), field_values)
+        return field_values
+
+    def clean(self, content_type, path=None, content=None, sheet_names=None):
```

Comment:
This function is in wip. I need to add the content parsing to it too. Working on it right now.